Simulator/Recorders/RecordableVector.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "RecordableBase.h"
+#include <stdexcept>   // for std::out_of_range
+#include <vector>
+// Cereal
+#include <cereal/types/polymorphic.hpp>
+#include <cereal/types/vector.hpp>
+
+template <typename T> class RecordableVector : public RecordableBase, public std::vector<T> {
+public:
```

Comment:
So, @KamathScience pointed out an issue with inheriting from `vector`. Is that resolved?

Reply:
@KamathScience Would you give a check to the serialization code in the RecordableVector class? Thanks.

Reply:
I tested the simulation on both CPU and GPU for serialization (test-small-connected and test-medium-connected) and these recent modifications work like a charm! Thank you @xiangli6